### PR TITLE
[dualtor][202012] Send arp/ndp packets from vlan_mac if the device is dualtor

### PR DESCRIPTION
**Pull request by vaibhavhd**
#### What I did

`fast-reboot-dump.pty` script collects fdb entries, will send out arp request for these fdb entries to keep them active longer.

However, the script sends these requests with switch mac.

Which has negative impacts on dualtor devices where the arp requests should come from vlan_mac, instead of router_mac

#### How I did it

Check the device subtype from metadata. If it is dualtor, use vlan_mac, otherwise use interface mac as the src address for ARP packets.

### CODE REVIEW

File: scripts/fast-reboot-dump.py [hunk 1]
```diff
@@ -266,6 +267,11 @@ def send_garp_nd(neighbor_entries, map_mac_ip_per_vlan):
     src_ifs = {map_mac_ip_per_vlan[vlan_name][dst_mac] for vlan_name, dst_mac, _ in neighbor_entries}
     src_mac_addrs = {src_if:get_iface_mac_addr(src_if) for src_if in src_ifs}
 
+    config_db = swsscommon.ConfigDBConnector()
+    config_db.connect()
+    device_metadata = config_db.get_table('DEVICE_METADATA')
+    subtype = device_metadata['localhost'].get('subtype', '')
```
Comment (yxieca) [inline scripts/fast-reboot-dump.py:273]: Is subtype case insensitive?

File: scripts/fast-reboot-dump.py [hunk 2]
```diff
@@ -275,10 +281,17 @@ def send_garp_nd(neighbor_entries, map_mac_ip_per_vlan):
     # send arp/ndp packets
     for vlan_name, dst_mac, dst_ip in neighbor_entries:
         src_if = map_mac_ip_per_vlan[vlan_name][dst_mac]
+        if subtype.lower() != 'dualtor':
```
Comment (yxieca) [inline scripts/fast-reboot-dump.py:284]: This is technically ok, but contains maybe safer?

Comment (vaibhavhd) [reply scripts/fast-reboot-dump.py]: Sure, addressed this now.

**Review by yxieca — APPROVED** [review]

File: scripts/fast-reboot-dump.py [hunk 3]
```diff
@@ -275,10 +281,17 @@ def send_garp_nd(neighbor_entries, map_mac_ip_per_vlan):
     # send arp/ndp packets
     for vlan_name, dst_mac, dst_ip in neighbor_entries:
         src_if = map_mac_ip_per_vlan[vlan_name][dst_mac]
+        if subtype and 'dualtor' in subtype.lower():
+            # for non dualtor devices, src mac will be router_mac
+            src_mac = src_mac_addrs[src_if]
+        else:
+            # for dualtor devices, the src mac will be vlan_mac
+            src_mac = get_iface_mac_addr(vlan_name)
```
Comment (prsunny) [inline scripts/fast-reboot-dump.py:289]: Isn't this the other way round?  This else case seems to be for non-dualtor.

Comment (vaibhavhd) [reply scripts/fast-reboot-dump.py]: Absolutely, yes :) Fixed this in latest. Finding out a way to test this change.

**Review by prsunny — APPROVED** [review]

**Review by yxieca — APPROVED** [review]

**Pull request merged**